src/App.tsx

```diff
@@ -1,16 +1,44 @@
 import React from 'react';
 import './App.scss';
+import { Counter } from './components/Counter';
 
-class App extends React.Component {
-  addOne = () => {};
+interface State {
+  count: number,
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    count: 0,
+  };
+
+  addOne = () => (
+    this.setState((previousState: State) => (
+      { count: previousState.count + 1 }))
+  );
+
+  add100 = () => (
+    this.setState((previousState: State) => (
+      { count: previousState.count + 100 }))
+  );
+
+  increase = () => {
+    const { count } = this.state;
 
-  add100 = () => {};
+    if (count % 5 === 0) {
+      this.add100();
+    }
 
-  increase = () => {};
+    this.addOne();
+  };
 
   render() {
     return (
-      <h1>Count: 0</h1>
+      <Counter
+        count={this.state.count}
```

Comment:
count - глагол
counter - существительное
```suggestion
        counter={this.state.counter}
```